src/main/java/org/photonvision/SimVisionSystem.java

```diff
@@ -0,0 +1,139 @@
+     * @param maxLEDRange_m Maximum distance at which your camera can illuminate the target and make it visible. Set to 9000 or more if your vision system does not rely on LED's.
+     * @param cameraRes_px Pair of integers representing the Width and Height (respectively) of your camera's image sensor in pixels
+     * @param minTargetArea Minimum area that that the target should be before it's recognized as a target by the camera. Match this with your contour filtering settings in the PhotonVision GUI.
+     */
+    public SimVisionSystem(String camName, double camDiagFOV_deg, double camPitch_deg, Transform2d robotToCamera, double cameraHeightOffGround_m, double maxLEDRange_m, Pair<Integer, Integer> cameraRes_px, double minTargetArea){
+        this.camDiagFOV_deg = camDiagFOV_deg;
+        this.camPitch_deg = camPitch_deg;
+        this.robotToCamera  = robotToCamera;
+        this.cameraHeightOffGround_m  = cameraHeightOffGround_m;
+        this.maxLEDRange_m = maxLEDRange_m;
+        this.cameraRes_px = cameraRes_px;
+        this.minTargetArea = minTargetArea;
+
+        //Calculate horizontal/vertical FOV by similar triangles
+        double hypotPixels = Math.hypot(cameraRes_px.getFirst(), cameraRes_px.getSecond());
+        this.camHorizFOV_deg = camDiagFOV_deg * cameraRes_px.getFirst()/hypotPixels;
+        this.camVertFOV_deg  = camDiagFOV_deg * cameraRes_px.getSecond()/hypotPixels;
+
+        cam = new SimPhotonCamera(camName);
+        tgtList = new LinkedList<SimVisionTarget>();
+    }
+
+    /**
+     * Add a target on the field which your vision system is designed to detect.
+     * The photoncamera from this system will report the location of the robot rellative
+     * to the subste of these targets which are visible from the given robot position.
+     * @param tgt
+     */
+    public void addSimVisionTarget(SimVisionTarget tgt){
+        tgtList.add(tgt);
+    }
+
+    /**
+     * Adjust the camera position relative to the robot.
+     * Use this if your camera is on a gimbal or turret or some other mobile platform.
+     * @param newRobotToCamera New Tranform from the robot to the camera
+     * @param newCamHeight New height of the camera off the floor
+     */
+    public void moveCamera(Transform2d newRobotToCamera, double newCamHeight){
+        this.robotToCamera  = newRobotToCamera;
+        this.cameraHeightOffGround_m  = newCamHeight;
+    }
+
+    /**
+     * Periodic update. Call this once per frame of image data you wish to process and send to NetworkTables
+     * @param robotPose current pose of the robot on the field. Will be used to calcualte which targets are actually in view, where they are at relative to the robot, and relevant PhotonVision parameters.
+     */
+    public void processFrame(Pose2d robotPose){
+
+        Pose2d cameraPos = robotPose.transformBy(robotToCamera);
+
+        LinkedList<PhotonTrackedTarget> visibleTgtList = new LinkedList<PhotonTrackedTarget>();
```

Comment:
why LinkedList? Just curious.

Reply:
Picked for constant-time add operation. However, thinking through it again, an `ArrayList` init'ed to the same size as the total number of possible targets would perform better. Assuming (I think) execution time is much more important than memory.